multicluster/apis/multicluster/v1alpha1/memberclusterannounce_webhook.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var memberclusterannouncelog = logf.Log.WithName("memberclusterannounce-resource")
+
+func (r *MemberClusterAnnounce) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/mutate-multicluster-crd-antrea-io-v1alpha1-memberclusterannounce,mutating=true,failurePolicy=fail,sideEffects=None,groups=multicluster.crd.antrea.io,resources=memberclusterannounces,verbs=create;update,versions=v1alpha1,name=mmemberclusterannounce.kb.io,admissionReviewVersions={v1,v1beta1}
+
+var _ webhook.Defaulter = &MemberClusterAnnounce{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *MemberClusterAnnounce) Default() {
+	memberclusterannouncelog.Info("default", "name", r.Name)
+}
+
+//+kubebuilder:webhook:path=/validate-multicluster-crd-antrea-io-v1alpha1-memberclusterannounce,mutating=false,failurePolicy=fail,sideEffects=None,groups=multicluster.crd.antrea.io,resources=memberclusterannounces,verbs=create;update,versions=v1alpha1,name=vmemberclusterannounce.kb.io,admissionReviewVersions={v1,v1beta1}
```

Comment:
`delete` is not part of the verbs to validate, is that intentional? if yes, then do we need the func definition at L61?

Reply:
I think the convention for Kubebuilder is that define every, and use based on need.

In this case, because delete request is not in the tag, it will not be forwarded to WebHooks, therefore L61 won't be called.